src/components/dashboard/products/ProductTable.tsx

```
@@ -0,0 +1,1127 @@
+
+  // ── 검색 입력 ─────────────────────────────────────────────────────────────
+  const { inputValue, onInputChange, onCompositionStart, onCompositionEnd } =
+    useSearchInput({ initialValue: filter.search, onSearch: handleSearch });
+
+  // ── 체크박스 선택 ─────────────────────────────────────────────────────────
+  const [selectedIds, setSelectedIds] = useState<Set<string>>(new Set());
+
+  // 필터 변경 시 선택 초기화
+  const prevFilterKey = useRef('');
+  const filterKey = `${filter.search}|${filter.status}|${filter.sort}|${filter.page}|${filter.limit}`;
+  if (prevFilterKey.current !== filterKey) {
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/components/dashboard/products/ProductTable.tsx:575:7
  573 |   const prevFilterKey = useRef('');
  574 |   const filterKey = `${filter.search}|${filter.status}|${filter.sort}|${filter.page}|${filter.limit}`;
> 575 |   if (prevFilterKey.current !== filterKey) {
      |       ^^^^^^^^^^^^^^^^^^^^^ Cannot access ref value during render
  576 |     prevFilterKey.current = filterKey;
  577 |     if (selectedIds.size > 0) setSelectedIds(new Set());
  578 |   }

(react-hooks/refs)